http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -115,6 +118,25 @@ private static Exception constructException(
         // Note: If another constructor is added, then we should refactor the construction logic in order to avoid
         // nested try/catch
         try {
+            if (exceptionClass.equals(ClientErrorException.class)
+                    || exceptionClass.equals(ServerErrorException.class)
+                    || exceptionClass.equals(WebApplicationException.class)) {
+                // the message should be of the form "HTTP $STATUS $DESCRIPTION
+                // for example, "HTTP 501 Not Implemented"
+                int prefixLength = "HTTP ".length();
+                String statusAsString = message.substring(prefixLength, prefixLength + 3);
```

Comment:
This may `throw IndexOutOfBoundsException` if the message doesn't contain "HTTP " or is length less than 5. Could also NPE if message is null.